.github/workflows/code-quality.yml

```diff
@@ -66,6 +52,5 @@ jobs:
           distribution: 'temurin'
           java-version: '17'
           cache: 'maven'
-      # install build tools first so changes to tool configuration is available downstream
-      - run: mvn -T1C -B -D skipTests -D skipChecks install -pl build-tools
-      - run: mvn -T1C -B -D skipTests -P checkFormat,-autoFormat validate
+      - run: mvn -T1C -B -D skipTests -D skipChecks install
```

Comment:
:question: This line is also part of CodeQL->Build. What do you think of doing this once in a prepare job and then making sure that java-checks and code-ql require this prepare step?

Reply:
There's no good option to share this work between multiple jobs. The only way is to upload and then download artifacts which is surprisingly slow. Having each job do a `mvn install` first is how we do it at the moment.